test/pg.ddl.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const metatests = require('metatests');
+const metaschema = require('metaschema');
+const { generateDDL } = require('../lib/pg.ddl');
+
+const defaultDomainsSql = `CREATE EXTENSION IF NOT EXISTS pgcrypto;
+
+
+-- Enum: "Month" ---------------------------------------------------------------
+
+CREATE TYPE "Month" AS ENUM (
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December'
+);
+
+-- Enum: "Day" -----------------------------------------------------------------
+
+CREATE TYPE "Day" AS ENUM (
+  'Monday',
+  'Tuesday',
+  'Wednesday',
+  'Thursday',
+  'Friday',
+  'Saturday',
+  'Sunday'
+);
+`;
+
+// Run test due to defined config
+//   config <Object> test config
+//     name <string> test name
+//     schemas <Object> schemas to use in test
+//     expectedSql <string> expected sql string
+const test = config => {
+  metaschema.fs.load(null, null, (err, schemas) => {
+    const test = metatests.test(config.name);
```

Comment:
This may result in multiple metatests 'finish' reports due to the fact that these tests are added asynchronously and if all other tests finish before this line gets called then metatests.Runner will assume that all tests have finished and print report.

You may add a file-global `const parentTest = metatestst.test('abc')` and use other tests as subtests of this `parentTest` or wait for https://github.com/metarhia/metatests/pull/122 (if it lands).

Reply:
IMHO, the best way to do it is to create all the subtest synchronously:
```js
const test = config =>
  metatests.test(config.name, test => {...load});
```

Moreoever, I'd inline this function.